Source/AwsCommonRuntimeKit/crt/CRTError.swift

```diff
@@ -1,12 +1,27 @@
 //  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 //  SPDX-License-Identifier: Apache-2.0.
-
-import Foundation
 import AwsCCommon
 
-public enum CRTError: Error {
-  case fileNotFound(String)
-  case memoryAllocationFailure
-  case stringConversionError(UnsafePointer<aws_string>?)
-  case crtError(AWSError)
+public struct CRTError: Error {
```

Comment:
Original question: what is the motivation behind not using specific errors? 

this is not ideal way to handle error in Swift. This is more like obj-c version where only one error type exists. This makes the code simple here but less than ideal for consumer.

A user friendly representation for something like https://github.com/awslabs/aws-c-sdkutils/blob/endpoint20-eval/include/aws/sdkutils/sdkutils.h#L26
```c
enum aws_sdkutils_errors {
    AWS_ERROR_SDKUTILS_GENERAL =    AWS_ERROR_ENUM_BEGIN_RANGE(AWS_C_SDKUTILS_PACKAGE_ID),
    AWS_ERROR_SDKUTILS_PARSE_FATAL,
    AWS_ERROR_SDKUTILS_PARSE_RECOVERABLE,
    AWS_ERROR_SDKUTILS_ENDPOINTS_UNSUPPORTED_RULESET,
    AWS_ERROR_SDKUTILS_ENDPOINTS_PARSE_FAILED,
    AWS_ERROR_SDKUTILS_ENDPOINTS_EVAL_INIT_FAILED,
    AWS_ERROR_SDKUTILS_ENDPOINTS_EVAL_FAILED,
    AWS_ERROR_SDKUTILS_ENDPOINTS_EMPTY_RULESET,
    AWS_ERROR_SDKUTILS_ENDPOINTS_RULESET_EXHAUSTED,

    AWS_ERROR_SDKUTILS_END_RANGE = AWS_ERROR_ENUM_END_RANGE(AWS_C_SDKUTILS_PACKAGE_ID)
};
```

would be
```swift
enum CRTSDKUtilsError {
    case general
    case parseFatal
    case parseRecoverable
    case endpointsUnsupportedRuleset
    case endpointsParseFailed
    case endpointsEvalInitFailed
    case endpointsEvalFailed
    case endpointsEmptyRuleset
    case endpointsRulesetExhausted
}

extension CRTSDKUtilsError {
    public var errorCode: Int32 {
        get {
            rawValue
        }
    }

    public var errorMessage: String? {
        get {
            guard let stringPtr = aws_error_str(errorCode) else {
                return nil
            }
            return String(cString: stringPtr)
        }
    }

    public var errorName: String? {
        get {
            guard let stringPtr = aws_error_name(errorCode) else {
                return nil
            }
            return String(cString: stringPtr)
        }
    }
}
```

Reply:
I just chatted with Waqar about this.

We have plans to do like you said use enums instead, but we're going to do it at a later stage. It's going to require doing some code-gen whenever we update the C submodules (scan through all the int error codes from the CRT, and create a nicely named swift enum for each one)

In the meantime we're trying to go from 3 different error classes (where it's not clear which one to use) down to 1

Reply:
I was trying to understand a little more this space and realized (I think) why there are 3 errors (1 is redundant)

Finally it should look like
```swift
/// Base enum for all CRT errors (Swift repo)
public enum CRTError: Error {
  case anyErrorThatPureSwiftCodeCanThrow
  case anyErrorThatPureSwiftCodeCanThrow2
  case awsError(AWSError) // case for all C errors
}

/// Base enum for all AWS CRT errors (C repo)
enum AWSError: Error {

}

Ideally we would want to have `enum` for `AWSError` but a `struct` is fine as of now
```